integrations/os2sync/lcdb_os2mo.py

```diff
@@ -0,0 +1,200 @@
+            break
+        branch.append(uuid)
+        lc_enhed = session.query(Enhed).filter(Enhed.uuid == uuid).one()
+        top_unit = uuid  # last one effective
+
+    # register top unit for all encountered
+    for buuid in branch:
+        top_per_unit[buuid] = top_unit
+    return top_unit
+
+
+def is_ignored(unit, settings):
+    """Determine if unit should be left out of transfer
+
+    Example:
+        >>> from unittest.mock import Mock
+        >>> unit=Mock(enhedsniveau_uuid="1", enhedstype_uuid="2")
+        >>> settings={
+        ... "OS2SYNC_IGNORED_UNIT_LEVELS": ["10","2"],
+        ... "OS2SYNC_IGNORED_UNIT_TYPES":['6','7']}
+        >>> is_ignored(unit, settings)
+        False
+        >>> unit.enhedstype_uuid="6"
+        >>> is_ignored(unit, settings)
+        True
+        >>> unit.enhedstype_uuid="2"
+        >>> is_ignored(unit, settings)
+        False
+        >>> unit.enhedsniveau_uuid="2"
+        >>> is_ignored(unit, settings)
+        True
+
+    Args:
+        unit: The organization unit to enrich with kle information.
+        settings: a dictionary
+
+    Returns:
+        Boolean
+    """
+
+    return (
+        unit.enhedstype_uuid in settings["OS2SYNC_IGNORED_UNIT_TYPES"] or
+        unit.enhedsniveau_uuid in settings["OS2SYNC_IGNORED_UNIT_LEVELS"])
+
+
+def get_sts_orgunit(session, uuid):
+    base = session.query(Enhed).filter(Enhed.uuid == uuid).one()
+
+    if is_ignored(base, settings):
+        return None
+
+    top_unit = get_top_unit(session, base)
+    if top_unit != settings["OS2MO_TOP_UNIT_UUID"]:
+        # not part of right tree
+        return None
+
+    sts_org_unit = {"ItSystemUuids": [], "Name": base.navn, "Uuid": uuid}
+
+    if base.forældreenhed_uuid is not None:
+        sts_org_unit["ParentOrgUnitUuid"] = base.forældreenhed_uuid
+
+    itconnections = session.query(ItForbindelse).filter(
+        ItForbindelse.enhed_uuid == uuid
+    ).all()
+    os2mo.itsystems_to_orgunit(
+        sts_org_unit,
+        [{"itsystem": {"uuid": itf.it_system_uuid}} for itf in itconnections]
+    )
+
+    addresses = []
+    for lc_address in session.query(Adresse).filter(
+        Adresse.enhed_uuid == uuid
+    ).all():
+        address = {
+            "address_type": {
+                "uuid": lc_address.adressetype_uuid,
+                "scope": scope_to_scope[lc_address.adressetype_scope]},
+            "name": lc_address.værdi,
+            "value": lc_address.dar_uuid,
+            "uuid": lc_address.uuid  # not used currently
+        }
+        addresses.append(address)
+    os2mo.addresses_to_orgunit(sts_org_unit, addresses)
+
+    mokles = {}
+    lc_kles = session.query(KLE).filter(KLE.enhed_uuid == uuid).all()
+    for lc_kle in lc_kles:
+        mokles.setdefault(lc_kle.uuid, {
+            "uuid": lc_kle.uuid,
+            "kle_aspect": []
+        })
+        mokles[lc_kle.uuid]["kle_aspect"].append(
+            {"scope": scope_to_scope[lc_kle.kle_aspekt_titel]}
+        )
```

Comment:
```suggestion
    import functools
    from collections import defaultdict
    def dict_constructor(out, lc_kle):
        out[lc_kle.uuid].append({"scope": scope_to_scope[lc_kle.kle_aspekt_titel]})
        return out
    lc_kles = session.query(KLE).filter(KLE.enhed_uuid == uuid).all()
    uuid_to_kles = functools.reduce(dict_constructor, lc_kles, defaultdict(list))
    mokles = {uuid: {'uuid': uuid, 'kle_aspect': kles} for uuid, kles in uuid_to_kles.items()}
```
Jeg ville nok lave denne for-løkke om til to operationer, en operation der bygger et uuid --> kle aspekt liste dict, og en der laver formatteringen. Den her set_default med et deltvist populated dictionary var meget svært for mig at fatte.